app/components/post_body_additional_content/post_body_additional_content.js

```diff
@@ -87,16 +88,24 @@ export default class PostBodyAdditionalContent extends PureComponent {
         }
     }
 
-    load = (props) => {
+    load = async (props) => {
         const {link} = props;
         if (link) {
             let imageUrl;
             if (isImageLink(link)) {
                 imageUrl = link;
-            } else if (isYoutubeLink(link)) {
-                const videoId = getYouTubeVideoId(link);
-                imageUrl = `https://i.ytimg.com/vi/${videoId}/hqdefault.jpg`;
-                ImageCacheManager.cache(null, `https://i.ytimg.com/vi/${videoId}/default.jpg`, () => true);
+            }
+
+            if (!imageUrl) {
+                imageUrl = await getShortenedLink(link);
+                if (imageUrl) {
+                    this.setState({shortenedLink: imageUrl});
+                    if (isYoutubeLink(imageUrl)) {
```

Comment:
Can we check if the link is a YouTube link before trying to follow the link as well?